client/components/checkbox/index.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+* External dependencies
+*/
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+* Interal dependencies
+*/
+import Gridicon from 'components/gridicon';
+
+export default React.createClass( {
+	displayName: 'Checkbox',
+
+	propTypes: {
+		checked: PropTypes.bool.isRequired,
+		disabled: PropTypes.bool,
+		onChange: PropTypes.func
+	},
+
+	getInitialState() {
+		return {
+			focused: false
+		}
+	},
+
+	getDefaultProps() {
+		return {
+			disabled: false
+		};
+	},
+
+	handleChange( e ) {
+		const { onChange } = this.props;
+
+		onChange && onChange( e, this.state.checked );
+	},
+
+	handleFocus() {
+		this.setState( {
+			focused: true
+		} )
+	},
+
+	handleBlur() {
+		this.setState( {
+			focused: false
+		} )
+	},
```

Comment:
concerning the change, blur, and focus functions - [we prefer to give descriptive names](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#react-method-names), so for instance, we could use these instead:

```js
toggleCheckmark( e ) {
	this.props.onChange( e, this.state.checked );
}

focus() {
	this.setState( { focused: true } );
}

blur() {
	this.setState( { focused: false } );
}

…

onChange={ this.toggleCheckmark }
onFocus={ this.focus }
onBlur={ this.blur }
```